gluten-core/src/main/scala/io/glutenproject/extension/columnar/TransformHintRule.scala

```diff
@@ -684,6 +684,9 @@ case class AddTransformHintRule() extends Rule[SparkPlan] {
               plan.isSkewJoin)
             TransformHints.tag(plan, transformer.doValidate().toTransformHint)
           }
+        case plan: CartesianProductExec =>
+          val transformer = CartesianProductExecTransformer(plan.left, plan.right, plan.condition)
```

Comment:
can you please also add a config to enable/disable CP here?

Reply:
Added config to enable/disable cp.

Reply:
Added backend specific change along with config